service/storage_proxy.cc

```diff
@@ -1655,6 +1656,7 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
             signal(from);
             using std::swap;
             swap(*it, _targets.back());
+            _successful_targets.push_back(std::move(_targets.back()));
```

Comment:
This makes me a bit anxious because it can throw due to an allocation error. While not marked as `noexcept`, much of the `abstract_write_response_handler` code is written in such a way that it doesn't throw. In particular, I'm not confident that storage proxy would gracefully handle this if an exception were thrown here.

How about pre-allocating the space in `_successful_targets` in the constructor of `abstract_write_response_handler`? It will have at most `_targets.size()` elements, so you could preallocate this much space.

Reply:
I removed `_successful_targets` and doing pre-calculation instead